typing_extensions/src_py3/typing_extensions.py

```diff
@@ -1220,8 +1224,10 @@ def _proto_hook(other):
                                 return NotImplemented
                             break
                         annotations = getattr(base, '__annotations__', {})
-                        if (isinstance(annotations, typing.Mapping) and attr in annotations and
-                                isinstance(other, _ProtocolMeta) and other._is_protocol):
+                        if (isinstance(annotations, typing.Mapping) and
```

Comment:
I think if all() may work here to cut down on the superfluous `and`'s

Reply:
I don't see how that could work since the `and` operator here is used for the benefit of its shortcut behavior (i.e. if the first condition fails, following conditions aren't evaluated). The only way that I can see `all()` used for something like `X and Y and Z` would be `all((X, Y, Z))` and that would evaluate all of `X`, `Y` and `Z` before calling `all()`.

Reply:
That's curious; I suppose the implication here is that all() does not implement shortcut behavior. I wonder if maybe it should? I'll see if I can explore this a bit.

Reply:
Well, actually, now I'm just unsure if maybe I'm not understanding your point gvanrossum@; to me, it looks like all() supports 'short-circuiting' See also: https://www.geeksforgeeks.org/short-circuiting-techniques-python/

Reply:
The tuple will be fully evaluated before it's passed into `all(...)`-- doing `all((X, Y, Z))` is basically the same thing as doing `temp = (X, Y, Z); all(temp)`. So, the short-circuiting isn't helpful here.

If we want to take advantage of `all(...)`'s short-circuiting behavior, we'd need to instead pass in some sort of iterator that computes and yield values on demand. For example, we could use generator comprehensions and do something like `if all(isinstance(x, str) for x in args): ...` -- generators are a kind of iterator. But in this case, the boolean conditions are heterogeous, which makes writing this kind of nicely compact generator comprehension pretty much impossible. (Give it a shot if you're not sure why).

More broadly, a good rule of thumb is that `all(...)` is useful primarily when you have some loop that checks whether every bool in some iterable is true and you want to compress that loop into a one-liner -- it's easy to rewrite simple loops as comprehensions. (And the corollary is that if you can't rewrite the checks you want to perform into some sort of loop, `all(...)` is unlikely to be helpful.)